config/config.go

```diff
@@ -1327,15 +1324,25 @@ func (c *Config) buildAggregator(name string, tbl *ast.Table) (*models.Aggregato
 		Grace:  time.Second * 0,
 	}
 
-	c.getFieldDuration(tbl, "period", &conf.Period)
-	c.getFieldDuration(tbl, "delay", &conf.Delay)
-	c.getFieldDuration(tbl, "grace", &conf.Grace)
-	c.getFieldBool(tbl, "drop_original", &conf.DropOriginal)
-	c.getFieldString(tbl, "name_prefix", &conf.MeasurementPrefix)
-	c.getFieldString(tbl, "name_suffix", &conf.MeasurementSuffix)
-	c.getFieldString(tbl, "name_override", &conf.NameOverride)
-	c.getFieldString(tbl, "alias", &conf.Alias)
-	c.getFieldString(tbl, "log_level", &conf.LogLevel)
+	found, period := c.getFieldDuration(tbl, "period")
+	if found {
+		conf.Period = period
+	}
+	found, delay := c.getFieldDuration(tbl, "delay")
+	if found {
+		conf.Delay = delay
+	}
+	found, grace := c.getFieldDuration(tbl, "grace")
+	if found {
+		conf.Grace = grace
+	}
+
+	conf.DropOriginal = c.getFieldBool(tbl, "drop_original")
+	conf.MeasurementPrefix = c.getFieldString(tbl, "name_prefix")
+	conf.MeasurementSuffix = c.getFieldString(tbl, "name_suffix")
+	conf.NameOverride = c.getFieldString(tbl, "name_override")
+	conf.Alias = c.getFieldString(tbl, "alias")
+	conf.LogLevel = c.getFieldString(tbl, "log_level")
```

Comment:
These can directly be done in the assignment like:
```go
	conf := &models.AggregatorConfig{
		Name:   name,
		Delay:  time.Millisecond * 100,
		Period: time.Second * 30,
		Grace:  time.Second * 0,
		DropOriginal: c.getFieldBool(tbl, "drop_original"),
	}
```

Reply:
I'm not convinced. The current version nicely shows the initialization of defaults first, followed by getting values parsed previously from the config.

Reply:
There aren’t defaults for the ones mentioned here…

Reply:
@srebhan what's your take on this?

Reply:
Let's keep this as-is for now as we cannot change this consistently in the file within the current PR!